tools/benchmark/src/main.rs

```diff
@@ -13,16 +13,51 @@ struct Input {
 
 const TERA: u64 = 1000000000000_u64;
 
-async fn bench_contract(wtr: &mut Writer<File>, name_os: OsString, commit: String) -> anyhow::Result<()> {
+const ASSERTATION_LOWER_BOUND: u64 = 230_u64;
+const ASSERTATION_UPPER_BOUND: u64 = 280_u64;
```

Comment:
Interesting! I didn't know there was such a large variance in the compiled artifact performance. I think it would be good to work on getting this variance down over time as well. Could we have the code here run the benchmark multiple times and compute a mean + variance to compare against? I worry that with such a wide range right now it will be hard to see improvements. For example if we go from producing a result anywhere in this 230 - 280 range to consistently producing 235 then our benchmarks should catch this.

Reply:
Yes, it really will improve representativeness of collected data. I added it [here ](https://github.com/MCJOHN974/evm2near/pull/18/commits/af2ba4990adb673288a2a0b7386cb19a53775366). Also, I removed some fields (with such big variance variables measured in gas are useless, only Tgas is important) and removed some all other benchmarking contracts to save workflow time and do more iterations. And, of course, now assertion is made on mean value. You can see results of last benchmarking with mean, variance, etc [here](https://github.com/MCJOHN974/evm2near/blob/gh-pages/csvs/af2ba49.csv)

Reply:
Also, workflow finished in just 9 minutes, so, maybe it will be reasonable increase amount of iterations to 50 for example and total workflow time will be not very terrible. I think it will be less than 20 minutes or something like that

Reply:
the main source of instability is dynamic edges translation to if-chain on that branch
it happens because ifs can be ordered a little bit different every time, which affects number of executed wasm instructions per cycle (comparing to overall (and quite small) number of instructions)
br_table branch will solve that problem (that is one of the reasons it was done first), but there is possibility of nonstable compilation affecting runtime costs in many ways, so we will dig each case on its own =)

Reply:
Maybe interesting, I three times generate sets of numbers from Gaussian distribution with mean=250 and variance=100. All sets was size = 20. And I calculated mean and variance for all three sets. Results:

```   
Mean = 252.16460983684718
Variance = 116.474188182477
Mean = 248.45788672714087
Variance = 62.67953851334892
Mean = 247.60003511533733
Variance = 104.08535484282378
```

It is very similar to our case, so, big spread between variance values is ok.